wwwroot/inc/interface-lib.php

```diff
@@ -1163,4 +1167,87 @@ function makeHtmlTag ($tagname, $attributes = array())
 	return $ret;
 }
 
+
+function colorHex2Rgb($color)
+{
+	$color = trim($color,'#');
+	$rgb = hexdec(substr($color,0,2)).",".hexdec(substr($color,2,2)).",".hexdec(substr($color,4,2));
+	return $rgb;
+}
+
+$objstyles_cache = array();
+function getObjectClass($realm, $object, $extrastyle = "")
+{
+	global $objstyles_cache;
+
+	$style = "";
+	switch($realm)
+	{
+		case 'object':
+			if(isset($object['colors'][0]))
+			{
+				$step = 100 / count($object['colors']);
+				$percent = 0;
+				$gradient = "";
+				foreach($object['colors'] as $color)
+				{
+					$rgb = colorHex2Rgb($color);
+					$gradient .= "rgba($rgb,0.2) $percent%, rgba($rgb,0.3) ".($percent + $step)."%,";
+					$percent += $step;
+				}
+
+				$style = "${extrastyle}background-image:linear-gradient(135deg,".trim($gradient,',').");";
+				$class = "tagcolor-".$object['id'];
+			}
+			break;
+		case 'tag':
+			if(isset($object['color']))
+			{
+				$rgb = colorHex2Rgb($object['color']);
+				$style = "${extrastyle}background: white linear-gradient(rgba($rgb,0.3), rgba($rgb,0.3));";
+
+				$class = "tag-".$object['id'];
+			}
+			break;
+	}
+
+	if($style != "")
```

Comment:
To me it seems this `if()` wrapper could go away if there was a simple `return ''` in a default case in the switch block above.